src/Verilator.cpp

```diff
@@ -609,5 +629,27 @@ int main(int argc, char** argv, char** env) {
 #endif
     FileLine::deleteAllRemaining();
 
+    if (v3Global.hierPlanp()) {
+        std::string cmd = "make ";
+        if (v3Global.opt.maxProc() > 1)
+            cmd += " -j" + cvtToStr(v3Global.opt.maxProc());
+        cmd += " -f " + v3Global.opt.prefix() + "_hier_verilation.mk ";
+        cmd += " -C " + v3Global.opt.makeDir();
+        cmd += v3Global.opt.doBuild() ? " all" : " verilate";
+        V3Os::system(cmd.c_str());
+    } else if (v3Global.opt.doBuild()) {
+        if (v3Global.opt.gmake()) {
+            std::string cmd = "make ";
+            if (v3Global.opt.maxProc() > 1)
+                cmd += " -j" + cvtToStr(v3Global.opt.maxProc());
+            cmd += " -f " + v3Global.opt.prefix() + ".mk ";
+            cmd += " -C " + v3Global.opt.makeDir();
+            V3Os::system(cmd.c_str());
+        } else if (v3Global.opt.cmake()) {
+            //V3EmitCMake::emitHierVerilation(v3Global.hierPlanp());
+            UASSERT(false, "Not implemented yet");
+        }
+    }
+
```

Comment:
Suggest separate into buildArguments and then a system call using the buildArguments.
Need to fail if the build returns bad exit status.